### PR TITLE
feat: add validation for notification data

### PR DESCRIPTION
**Pull request by LakshayaT**
This PR adds validation to the notification data, when sent via API call and also before adding the notification ID into queue.

Related changes:
- Create a DTO for notification data, add rules
- Update notification dto, entity to change type of data to CreateNotificationDataDto Update notification service to validate notification data from database via the new DTO before adding to queue

Sample Erroneous Request:
```json
{
  "channelType": 1,
  "data": {
    "to": "receiver@email.com",
    "subject": "Test Notification",
    "text": "This is a test notification",
    "html": "<b>This is a test notification</b>"
  }
}
```

Sample Response:
```json
{
  "message": [
    "data.from should not be empty"
  ],
  "error": "Bad Request",
  "statusCode": 400
}
```

Error Logging in Console:
```bash
[OsmoNotify] Error      9/7/2023, 6:45:00 PM [NotificationsService] Error adding notification with id: 9 to queue - {"stack":[null]} +9ms
[OsmoNotify] Error      9/7/2023, 6:45:00 PM [NotificationsService] [
  {
    "property": "from",
    "children": [],
    "constraints": {
      "isNotEmpty": "from should not be empty"
    }
  }
] - {"stack":[null]} +1ms
```

### CODE REVIEW

File: src/modules/notifications/notifications.service.ts [hunk 1]
```diff
@@ -46,11 +49,15 @@ export class NotificationsService {
...
+        await validateOrReject(plainToClass(CreateNotificationDataDto, notification.data), {
+          validationError: { target: false },
+        });
         await this.notificationQueueService.addNotificationToQueue(notification);
       } catch (error) {
         notification.deliveryStatus = DeliveryStatus.PENDING;
+        notification.result = { result: error };
         this.logger.error(`Error adding notification with id: ${notification.id} to queue`);
-        this.logger.error(JSON.stringify(error, ['message', 'stack'], 2));
+        this.logger.error(JSON.stringify(error, null, 2));
```
Comment (osm-vishnukyatannawar) [inline src/modules/notifications/notifications.service.ts:60]: Why this change?

Did you check if adding `null` is printing stack and message?

**Review by osm-vishnukyatannawar — CHANGES_REQUESTED** [review]

**Comment by osm-vishnukyatannawar** [comment]
Is attachment also validated?

File: src/modules/notifications/notifications.service.ts (hunk 1 above)
Comment (LakshayaT) [reply src/modules/notifications/notifications.service.ts:60]: When only `['message', 'stack']` is specified, the error is logged as follows:
image

`null` was used instead due to the way the `error` value itself is generated. It does not have a `message` or `stack` value:
image

And when using `null`, error is logged as follows:
image

Comment (osm-vishnukyatannawar) [reply src/modules/notifications/notifications.service.ts:60]: Okay for now. But I thought it was done in that specific way for a reason before.

**Review by osm-vishnukyatannawar — APPROVED** [review]
